src/Service/Search/SolrSelect.php

```diff
@@ -0,0 +1,313 @@
+            $solrQuery,
+            $query->defaultQueryOperator
+        );
+        $this->addFilterQueriesToSolrQuery(
+            $solrQuery,
+            $query->filter
+        );
+        $this->addFacetListToSolrQuery(
+            $solrQuery,
+            $query->facets
+        );
+
+        return $solrQuery;
+    }
+
+    /**
+     * @param Criteria[] $criteriaList
+     */
+    private function addSortToSolrQuery(
+        SolrSelectQuery $solrQuery,
+        array $criteriaList
+    ): void {
+        $sorts = [];
+        foreach ($criteriaList as $criteria) {
+            if ($criteria instanceof Name) {
+                $field = 'sp_name';
+            } elseif ($criteria instanceof Headline) {
+                $field = 'sp_title';
+            } elseif ($criteria instanceof Date) {
+                $field = 'sp_date';
+            } elseif ($criteria instanceof Natural) {
+                $field = 'sp_sortvalue';
+            } elseif ($criteria instanceof Score) {
+                $field = 'score';
+            } else {
+                throw new InvalidArgumentException(
+                    'unsupported sort criteria: ' . get_class($criteria)
+                );
+            }
+
+            $direction = strtolower($criteria->direction->name);
+
+            $sorts[$field] = $direction;
+        }
+        $solrQuery->setSorts($sorts);
+    }
+
+    private function addRequiredFieldListToSolrQuery(
+        SolrSelectQuery $solrQuery
+    ): void {
+        $solrQuery->setFields(['url']);
+    }
+
+    private function addTextFilterToSolrQuery(
+        SolrSelectQuery $solrQuery,
+        string $text
+    ): void {
+        if (empty($text)) {
+            return;
+        }
+        $terms = explode(' ', $text);
+        $terms = array_map(
+            static function ($term) use ($solrQuery) {
+                $term = trim($term);
+                return $solrQuery->getHelper()->escapeTerm($term);
+            },
+            $terms
+        );
+        $text = implode(' ', $terms);
+        $solrQuery->setQuery($text);
+    }
+
+    private function addQueryDefaultOperatorToSolrQuery(
+        SolrSelectQuery $solrQuery,
+        QueryOperator $operator
+    ): void {
+        if ($operator === QueryOperator::OR) {
+            $solrQuery->setQueryDefaultOperator(
+                SolrSelectQuery::QUERY_OPERATOR_OR
+            );
+        } else {
+            $solrQuery->setQueryDefaultOperator(
+                SolrSelectQuery::QUERY_OPERATOR_AND
+            );
+        }
+    }
+
+    /**
+     * @param Filter[] $filterList
+     */
+    private function addFilterQueriesToSolrQuery(
+        SolrSelectQuery $solrQuery,
+        array $filterList
+    ): void {
+
+        foreach ($filterList as $filter) {
+            $key = $filter->key ?? uniqid('', true);
+            $solrQuery->createFilterQuery($key)
+                ->setQuery($filter->getQuery())
+                ->setTags($filter->tags);
```

Comment:
same as `SolrMoreLikeThis`

```suggestion
            $filterQuery = $solrQuery->createFilterQuery($key);
            $filterQuery->setQuery($filter->getQuery());
            $filterQuery->setTags($filter->tags);
```